File: option/src/app/node_mysql/src/rest.js
const express = require('express')
const app = express()
const port = 8080

const mysql = require('mysql2');

app.get('/info', (req, res) => {
    res.send('NodeJS - Express / MySQL')
})

app.get('/dept', (req, res) => {
    const aDbURL= process.env.DB_URL.split(":");
    var con = mysql.createConnection({
        host: aDbURL[0],
        user: process.env.DB_USER,
        password: process.env.DB_PASSWORD,
        database: "db1"
    });

    con.connect();
    
    con.query("SELECT deptno, dname, loc FROM dept", function (err, result, field) {
        if (err) throw err;
        console.log(result);
        res.send(result)
      });

      con.end();
})

app.listen(port, () => {
    console.log(`OCI Starter: listening on port ${port}`)
})

File: option/src/app/python_mysql/src/rest.py
import os 
from app import app
from flask import jsonify
import mysql.connector

@app.route('/dept')
def dept():
    try:
        conn = mysql.connector.connect(**config)
        cursor = conn.cursor()
        cursor.execute("SELECT deptno, dname, loc FROM dept")
        deptRows = cursor.fetchall()
        a = []
        for row in deptRows:
            a.append( {"deptno": row[0], "dname": row[1], "loc": row[2]} )
        print(a)
        response = jsonify(a)
        response.status_code = 200
        return response
    except Exception as e:
        print(e)
    finally:
        cursor.close() 
        conn.close()     

@app.route('/info')
def info():
        return "Python - Flask / MySQL"          

db_url = os.getenv('DB_URL')
mysql_host = db_url.split(':')[0]
config = {
    "host": mysql_host,
    "port": 3306,
    "database": "db1",
    "user": os.getenv('DB_USER'),
    "password": os.getenv('DB_PASSWORD'),
    "charset": "utf8",
    "use_unicode": True,
    "get_warnings": True
}

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)